src/mscorlib/src/System/Globalization/HijriCalendar.Win32.cs

```diff
@@ -42,54 +42,41 @@ private static int GetAdvanceHijriDate()
             int hijriAdvance = 0;
             Microsoft.Win32.RegistryKey key = null;
 
-            try
+            using (key = Registry.CurrentUser.OpenSubKey(InternationalRegKey, writable: false))
             {
-                // Open in read-only mode.
-                // Use InternalOpenSubKey so that we avoid the security check.
-                key = RegistryKey.GetBaseKey(RegistryKey.HKEY_CURRENT_USER).OpenSubKey(InternationalRegKey, false);
-            }
-            //If this fails for any reason, we'll just return 0.
-            catch (ObjectDisposedException) { return 0; }
-            catch (ArgumentException) { return 0; }
```

Comment:
this could be a behavior change as we used to continue if we get ObjectDisposedException exceptions but now we'll throw.